libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1779,4 +1886,22 @@ export class RushConfiguration {
       ...(variant ? [RushConstants.rushVariantsFolderName, variant] : [])
     );
   }
+
+  /**
+   * The workspace project can NOT depend on a split workspace project.
+   * The split workspace project CAN depend on a workspace project.
+   */
+  private _validateSplitWorkspaceRelationships(): void {
+    for (const project of this.projects) {
+      if (!project.splitWorkspace) {
+        for (const dependencyProject of project.dependencyProjects) {
+          if (dependencyProject.splitWorkspace) {
+            throw new Error(
+              `Split workspace project ${dependencyProject.packageName} is not allowed to be dependency of project ${project.packageName}`
```

Comment:
Since the fix is in `project`, it might be more intuitive to mention it first in the message:

`The "${project.packageName}" cannot depend on "${dependencyProject.packageName}" because regular workspace projects cannot depend on projects from the split workspace.`